tests/test_tuya_dimmer.py

```diff
@@ -0,0 +1,256 @@
+        b"\tV\x02\x01y\x08\x02\x00\x04\x00\x00\x02\x91"
+    )
+    tuya_cluster.handle_message(hdr, args)
+    succ, fail = await dimmer2_cluster.read_attributes(("current_level",))
+    assert succ["current_level"] == 167
+
+    # payload=400
+    hdr, args = tuya_cluster.deserialize(
+        b"\tV\x02\x01y\x08\x02\x00\x04\x00\x00\x01\x90"
+    )
+    tuya_cluster.handle_message(hdr, args)
+    succ, fail = await dimmer2_cluster.read_attributes(("current_level",))
+    assert succ["current_level"] == 102
+
+    # payload=149
+    hdr, args = tuya_cluster.deserialize(
+        b"\tV\x02\x01y\x08\x02\x00\x04\x00\x00\x00\x95"
+    )
+    tuya_cluster.handle_message(hdr, args)
+    succ, fail = await dimmer2_cluster.read_attributes(("current_level",))
+    assert succ["current_level"] == 37
+
+    # payload=339
+    hdr, args = tuya_cluster.deserialize(
+        b"\tV\x02\x01y\x08\x02\x00\x04\x00\x00\x01\x53"
+    )
+    tuya_cluster.handle_message(hdr, args)
+    succ, fail = await dimmer2_cluster.read_attributes(("current_level",))
+    assert succ["current_level"] == 86
+
+
+@pytest.mark.parametrize(
+    "quirk", (zhaquirks.tuya.ts0601_dimmer.TuyaDoubleSwitchDimmer,)
+)
+async def test_doubledimmer_state_report(zigpy_device_from_quirk, quirk):
+    """Test tuya double switch."""
+
+    # TUYA_EP2_SWITCH_ON = b"\tG\x02\x01o\x07\x01\x00\x01\x01"
+    # TUYA_EP2_SWITCH_OFF = b"\tH\x02\x01p\x07\x01\x00\x01\x00"
+
+    TUYA_EP2_DIMM_1 = b"\tV\x02\x01y\x08\x02\x00\x04\x00\x00\x02\xc5"
+    TUYA_EP2_DIMM_2 = b"\tW\x02\x01z\x08\x02\x00\x04\x00\x00\x02\x9e"
+
+    dimmer_dev = zigpy_device_from_quirk(quirk)
+
+    dimmer1_cluster = dimmer_dev.endpoints[1].level
+    dimmer1_listener = ClusterListener(dimmer1_cluster)
+
+    dimmer2_cluster = dimmer_dev.endpoints[2].level
+    dimmer2_listener = ClusterListener(dimmer2_cluster)
+
+    tuya_cluster = dimmer_dev.endpoints[1].tuya_manufacturer
+
+    assert len(dimmer1_listener.cluster_commands) == 0
+    assert len(dimmer1_listener.attribute_updates) == 0
+    assert len(dimmer2_listener.cluster_commands) == 0
+    assert len(dimmer2_listener.attribute_updates) == 0
+
+    # events from channel 2 updates only EP 2
+    hdr, args = tuya_cluster.deserialize(TUYA_EP2_DIMM_1)
+    tuya_cluster.handle_message(hdr, args)
+    assert len(dimmer1_listener.attribute_updates) == 0
+    assert len(dimmer2_listener.attribute_updates) == 1
+    assert dimmer2_listener.attribute_updates[0][0] == 0x0000
+    assert dimmer2_listener.attribute_updates[0][1] == 180
+
+    # events from channel 1 updates only EP 1
+    hdr, args = tuya_cluster.deserialize(TUYA_EP2_DIMM_2)
+    tuya_cluster.handle_message(hdr, args)
+    assert len(dimmer1_listener.attribute_updates) == 0
+    assert len(dimmer2_listener.attribute_updates) == 2
+    assert dimmer2_listener.attribute_updates[1][0] == 0x0000
+    assert dimmer2_listener.attribute_updates[1][1] == 170
+
+    # # events from channel 2 updates only EP 2
```

Comment:
These should be uncommented and made to work or they should be removed from the PR.

Reply:
Done